src/tools/annotation/FreehandMouseTool.js

```diff
@@ -1604,14 +1607,29 @@ export default class FreehandMouseTool extends BaseAnnotationTool {
   /**
    * Fire cornerstonetoolsmeasurementmodified event on provided element
    * @param {any} element which freehand data has been modified
-   * @param {any} data the measurment data
+   * @param {any} measurementData the measurment data
+   * @returns {void}
    */
-  fireModifiedEvent(element, data) {
-    external.cornerstone.triggerEvent(
+  fireModifiedEvent(element, measurementData) {
+    const eventType = EVENTS.MEASUREMENT_MODIFIED;
+    const eventData = {
+      toolName: this.name,
       element,
-      EVENTS.MEASUREMENT_MODIFIED,
-      data
-    );
+      measurementData,
+    };
```

Comment:
NOTE: the shape of the data was different here to every other tool which fires a `MEASUREMENT_MODIFIED` event. This is potentially a breaking change 😞or maybe its a bug fix 🤔?

Reply:
This looks the right structure to me. comparing the structure you have with v2.4 is similar. I would consider the old format a bug 
https://github.com/cornerstonejs/cornerstoneTools/blob/v2.4.x/src/imageTools/freehand.js#L1195-L1204